server/src/main/java/org/elasticsearch/node/NodeConstruction.java

```diff
@@ -415,8 +415,8 @@ private Settings createEnvironment(Environment initialEnvironment, NodeServicePr
             Constants.OS_ARCH,
             Constants.JVM_VENDOR,
             Constants.JVM_NAME,
-            Constants.JAVA_VERSION,
-            Constants.JVM_VERSION
+            System.getProperty("java.version"),
+            Runtime.version().toString()
```

Comment:
I'm a bit surprised these changes are not on "main", they don't seem Lucene specific?

Reply:
I was surprised too by this, I think it's ok though :)